app/controllers/api_controller.rb

```diff
@@ -1,9 +1,12 @@
 class APIController < ActionController::API
-  include EmitsRequestEvents
+  include DfE::Analytics::Requests
   include ActionController::HttpAuthentication::Token::ControllerMethods
   include Pundit::Authorization
 
   # child must define authenticate method
   before_action :authenticate
   after_action :verify_authorized
+
+  # Used by dfe-analytics to (try and) log the user triggering an event.
+  def current_user; end
```

Comment:
This is the tech debt that stems from a 3rd party dependency that we have control with. It's also fostering bad practice.

Can we just update `dfe-analytics`?

Solve the upstream so we don't pollute the downstream. :skull_and_crossbones: :ocean:

Reply:
I was going with the advice given in the `dfe-analytics` gem which was to provide an empty `current_user` method:

https://github.com/DFE-Digital/dfe-analytics#controllers

```ruby
  # This method MUST be present in your controller and should return
  # either nil or an object implementing an .id method.
  #
  # def current_user; end
```

Can we just merge this in for now and look to fix later? I'm happy to have a convo with @duncanjbrown about whether this really is the right approach.

Reply:
PR merge, will cut a new release shortly.